### DISCUSSION TITLE
Keyboard-driven URL opening

### DISCUSSION THREAD

**Discussion by jvoisin** [ghostty-org/ghostty#3922] [Feature Requests, Ideas] [answered]
[foot](https://codeberg.org/dnkl/foot) has a pretty [nice feature](https://codeberg.org/dnkl/foot/#urls) to deal with url:

> Foot supports URL detection. But, unlike many other terminal emulators, where URLs are highlighted when they are hovered and opened by clicking on them, foot uses a keyboard driven approach.

Pressing ctrl+shift+o enters "URL mode", where all currently visible URLs are underlined, and is associated with a "jump-label". The jump-label indicates the key sequence (e.g. "AF") to use to activate the URL.

It looks like this:

![image](https://github.com/user-attachments/assets/c1e9d5d8-952c-4ad6-a17c-cd00ce79520b)

It would be neat to have it in ghostty as well I think.

**Comment by NeverL4nd2** [comment]
I have same question, I use `wezterm`:

```lua
	{
		key = "u",
		mods = ...,
		action = wezterm.action.QuickSelectArgs({
			label = "open url",
			patterns = {
				"\\((https?://\\S+)\\)",
				"\\[(https?://\\S+)\\]",
				"\\{(https?://\\S+)\\}",
				"<(https?://\\S+)>",
				"\\bhttps?://\\S+[)/a-zA-Z0-9-]+",
				"\\b\\w+://(?:[\\w.-]+)\\.[a-z]{2,15}\\S*[^\\s.,;:]",
				"\\b\\w+@[\\w-]+\\.[\\w.-]+\\b",
			},
			action = wezterm.action_callback(function(window, pane)
				local url = window:get_selection_text_for_pane(pane)
				wezterm.open_with(url)
			end),
		}),
	},
```

**Comment by sjbarag** [comment]
Docs from some other terminals, in case that's helpful as a reference:

* Alacritty: [Hints](https://alacritty.org/config-alacritty.html#hints); default is open
* Kitty: Also [Hints](https://sw.kovidgoyal.net/kitty/kittens/hints/); default is open
* Foot: [URL mode](https://codeberg.org/dnkl/foot#keyboard); default is open
* WezTerm: [Quick Select Mode](https://wezfurlong.org/wezterm/quickselect.html); default is copy
* Konsole: [URL Hints](https://mail.kde.org/pipermail/konsole-devel/2018-September/030837.html) (unfortunately that mailing list post is the closest I can find to official documentation, besides [this reddit post](https://www.reddit.com/r/kde/comments/xi6m28/holding_down_alt_in_konsole_gives_me_these_number/) with a screenshot)

This obviously isn't an exhaustive list, since there are Too Many Terminals 😛

**Comment by sjbarag** [comment, marked as answer]
This seems to be a duplicate of the earlier https://github.com/ghostty-org/ghostty/discussions/2394

**Reply by jvoisin** [reply]
Oh, it is indeed, my bad!
